app/main.py

```diff
@@ -1,2 +1,119 @@
+from typing import Any, Hashable
+from fractions import Fraction
+from app.node import Node
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._length: int = 0
+        self._hash_table: list = [None] * 8
+        self._hash_size: int = 8
+        self._resize_factor: Fraction = Fraction(2, 3)
+        self._resize_multiplier: int = 2
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self._hash_size_check_increase()
+        key_exists, key_hash_position = self._while_key_hash_occupied(
+            action="set", key=key, value=value
+        )
+        if not key_exists:
+            node = Node(hash_=hash(key), key=key, value=value)
+            self._hash_table[key_hash_position] = node
+            self._length += 1
+
+    def __getitem__(self, key: Hashable) -> Any | None:
+        key_exists, return_value = self._while_key_hash_occupied(
+            action="get", key=key
+        )
+        if key_exists:
+            return return_value
+        raise KeyError("Key is not found in the Dictionary")
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __repr__(self) -> str:
+        result = [f"{item.key}:{item.value}"
+                  for item in self._hash_table
+                  if item is not None]
+        return "{" + ", \n".join(result) + "}"
+
+    def __delitem__(self, key: Hashable) -> None:
+        key_hash_position = self._find_key_position(key)
+        index_len = 0
+
+        while index_len <= self._hash_size:
+            if (self._hash_table[key_hash_position]
+                    and self._hash_table[key_hash_position].key == key):
+                self._hash_table[key_hash_position] = None
+                self._length -= 1
+                return
+
+            key_hash_position += 1
+            index_len += 1
+
+        raise KeyError("Key is not found in the Dictionary")
+
+    def pop(self, key: Hashable) -> Any:
+        result = self[key]
+        del self[key]
+        return result
```

Comment:
thats how pop method should work )
![image](https://github.com/mate-academy/py-dictionary/assets/103201010/9320ade5-1bc7-4f2a-9e2e-9d5e7a81178c)

Reply:
> thats how pop method should work ) ![image](https://private-user-images.githubusercontent.com/103201010/322375404-9320ade5-1bc7-4f2a-9e2e-9d5e7a81178c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxNjQwMjMsIm5iZiI6MTcxMzE2MzcyMywicGF0aCI6Ii8xMDMyMDEwMTAvMzIyMzc1NDA0LTkzMjBhZGU1LTFiYzctNGYyYS05ZTJlLTlkNWU3YTgxMTc4Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNVQwNjQ4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kODVjYmY4ODI2ZDAwZTFhMGI5ZWVmZTE0MmM1MjdkYzhmOTQ4YjNkYTA4OTFiNmZlNDUzNzA4MjM1ZjE1NWMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gt9k5plv5hMonMDXqKWnwAXCvC4wqomYXTahhBVCdE8)

you mean that I need to return None in my case instead of key error? so like ```result = self.get(key)``` instead of ```result = self[key]```?